PKHeX.WinForms/Program.cs

```diff
@@ -29,16 +29,33 @@ private static void Main()
 
             try
             {
-                if (GetFrameworkVersion() >= 393295)
+                if (IsOnWindows())
                 {
-                    StartPKHeX();
+                    if (GetFrameworkVersion() >= 393295)
+                    {
+                        StartPKHeX();
+                    }
+                    else
+                    {
+                        // Todo: make this translatable
+                        MessageBox.Show(".NET Framework 4.6 needs to be installed for this version of PKHeX to run.", "PKHeX Error", MessageBoxButtons.OK, MessageBoxIcon.Stop);
+                        Process.Start(@"https://www.microsoft.com/download/details.aspx?id=48130");
+                    }
                 }
                 else
                 {
-                    // Todo: make this translatable
-                    MessageBox.Show(".NET Framework 4.6 needs to be installed for this version of PKHeX to run.", "PKHeX Error", MessageBoxButtons.OK, MessageBoxIcon.Stop);
-                    Process.Start(@"https://www.microsoft.com/download/details.aspx?id=48130");
+                    //CLR Version 4.0.30319.42000 is equivalent to .NET Framework version 4.6
+                    if ((Environment.Version.CompareTo(Version.Parse("4.0.30319.42000"))) >= 0)
```

Comment:
Microsoft discourage the use of Environment Version to check for the .NET Framework Version as the CLR version doesn't necessary have to change. It just happened to be a clear version number for 4.6: https://msdn.microsoft.com/en-us/library/hh925568(v=vs.110).aspx#clr_b 

As I am only able to check on Arch (kernel 4.9.5) and Win 10, I chose the safer route.